app/pages/capabilities-selector/controller.test.js

```diff
@@ -6,8 +6,42 @@ jest.mock('../../apiProvider', () => ({
   getData: jest.fn(),
 }));
 
+const mockedCapabilitiesData = {
+  capabilities: [{
+    reference: 'C5',
+    version: '1.0.1',
+    name: 'Appointments Management - GP',
+    isFoundation: true,
+  }, {
+    reference: 'C13',
+    version: '1.0.1',
+    name: 'Patient Information Maintenance',
+    isFoundation: true,
+  }, {
+    reference: 'C14',
+    version: '1.0.1',
+    name: 'Prescribing',
+    isFoundation: true,
+  }],
+};
+
 describe('capabilities-selector controller', () => {
   describe('getCapabilitiesContext', () => {
+    afterEach(() => {
+      apiProvider.getData.mockReset();
+    });
+
+    it('should call getData once with the correct params', async () => {
+      apiProvider.getData
+        .mockResolvedValueOnce(mockedCapabilitiesData);
+
+      await getCapabilitiesContext();
+      expect(apiProvider.getData.mock.calls.length).toEqual(1);
+      expect(apiProvider.getData.mock.calls[0].length).toEqual(1);
+      expect(Object.keys(apiProvider.getData.mock.calls[0][0]).length).toEqual(1);
+      expect(apiProvider.getData.mock.calls[0][0].endpointLocator).toBe('getCapabilities');
```

Comment:
You are going to have to explain this one to me on Monday

Reply:
so `apiProvider.getData.mock.calls` is an array of the calls made to the mock
`apiProvider.getData.mock.calls[0]` is an array of the args used when the mock was called
`apiProvider.getData.mock.calls[0][0]` is one of the args

Reply:
https://jestjs.io/docs/en/mock-functions.html

Reply:
so it just checks that the getData function is being called with the correct args (which is what changed and made the bug but all the tests still passed - if it happens again the tests will fail)

Reply:
Right I get this now. It kinda looks pretty difficult to get you head round looking into it the first time. Can anything be done to make it read a little better. I noticed in the link you sent you can use Custom Matchers 
`// The mock function was called at least once with the specified args
expect(mockFunc).toHaveBeenCalledWith(arg1, arg2);`
can something like this be used to check getData is being called with the correct args?

Reply:
ooh yeah that would be better - let me look a bit more